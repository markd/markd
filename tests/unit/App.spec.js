import { shallowMount, createLocalVue } from '@vue/test-utils'
import Component from '@/App.vue'

import Vuex from 'vuex'
import VueRouter from 'vue-router'

const localVue = createLocalVue()

localVue.use(Vuex)
localVue.use(VueRouter)

const router = new VueRouter()

const store = new Vuex.Store({
  state: {
    title: 'markd',
    pinned: [
      {
        icon: 'cloud_queue',
        name: 'All Bookmarks'
      },
      {
        icon: 'favorite_border',
        name: 'Favourite Bookmarks'
      }
    ],
    collections: [
      {
        colour: 'red',
        name: 'Web Development'
      },
      {
        colour: 'green',
        name: 'Spicy Memes'
      }
    ]
  },
  actions: {
    initialLoad() {}
  }
})

describe('App.vue', () => {
  it('renders', () => {
    const wrapper = shallowMount(Component, {
      localVue,
      store,
      router
    })
    expect(wrapper).toMatchSnapshot()
  })
})
