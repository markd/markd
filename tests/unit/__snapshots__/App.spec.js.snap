// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue renders 1`] = `
<div id="app" class="light">
  <icons-stub></icons-stub>
  <navigation-stub></navigation-stub>
  <sidebar-stub></sidebar-stub>
  <main class="content">
    <router-view-stub name="default"></router-view-stub>
  </main>
</div>
`;
